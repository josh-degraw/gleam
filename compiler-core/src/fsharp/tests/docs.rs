use crate::assert_fsharp;

#[test]
fn single_line_doc_comment() {
    assert_fsharp!(
        r#"
/// This is a doc comment
fn other() {
  Nil
}

pub fn foo() {
  other
}
"#
    );
}

#[test]
fn multi_line_doc_comment() {
    assert_fsharp!(
        r#"
/// This is a doc comment
/// on multiple lines
fn other() {
  Nil
}

pub fn foo() {
  other()
}
"#
    );
}

#[test]
fn deprecated_function_with_doc_comment() {
    assert_fsharp!(
        r#"
/// This is a doc comment
/// on multiple lines
@deprecated("Use new_function instead")
fn other() {
  Nil
}

pub fn foo() {
  other()
}
"#
    );
}
