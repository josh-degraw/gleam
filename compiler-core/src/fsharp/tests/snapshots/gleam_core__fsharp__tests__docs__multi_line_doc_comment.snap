---
source: compiler-core/src/fsharp/tests/docs.rs
expression: "\n/// This is a doc comment\n/// on multiple lines\nfn other() {\n  Nil\n}\n\npub fn foo() {\n  other()\n}\n"
---
module rec my.``mod``

/// This is a doc comment
/// on multiple lines
let private other () = begin
    ()
end
let foo () = begin
    other()
end
